fault/system_verilog_target.py

```diff
@@ -722,15 +722,19 @@ def generate_code(self, actions, power_args):
         # return the string representing the system-verilog testbench
         return src
 
-    def run(self, actions, power_args=None):
+    def generate_test_bench(self, actions, power_args=None):
```

Comment:
looks like a different interface? i.e. one takes in actions one takes in target? It does seem a bit weird to have to pass in target into compile and then generate_test_bench.

What if you do

```python
tester.compile("target1", ...)
tester.generate_test_bench("target2", ...)
```

Reply:
If you haven't compiled `target2`, you'll get an error (https://github.com/leonardt/fault/pull/271/files#diff-39f4c94be89b8849a9043859f74ec7aef421fcb4ff22b3d08c629a82e89f993cR239-R244)
If you have compiled `target2` it will work fine.

The interface supports multiple compilation targets, although I don't think anyone is using this feature.  I think the main reason to separate `compile` from `run` (and now generate_test_bench) was that we wanted to support pre compiling the verilator model to save time in test runs, so some tests just call generate_test_bench or run.  If we didn't want to support multiple compilation targets for a test, we could then attach a specific target and not have to specify the targets for generate_test_bench/run

Reply:
Also the targets have a different interface than the tester because the tester passes it's actions through to the target (so the target API isn't really user facing, it's meant to be invoked via the tester API)

Reply:
Oops my mistake on target vs tester classes.

Got it. Would argue in general this pattern could be a little non-robust if you do weird stuff like more actions in between compile and run or something. Or call things in weird orders. But that's definitely "user error".